### ISSUE TITLE
quality/duplicate-dict-key triggers on enums from other classes

### ISSUE THREAD

**Issue by EvilsPresley** [atelico/gdstyle#8]
### Minimal reproduction

```gdscript
2 files:

-----
class_name Enums
extends Node

enum E {
	KEY_A,
}

-----
extends Node

const _DEFS: Dictionary = {
	Enums.E.KEY_A: {
		"value_1": "A",
	},
}
```

### Command you ran

gdstyle check

### What gdstyle did

>   4:9 warning duplicate dictionary key '.' (first seen at line 4) [quality/duplicate-dict-key]
>       	Enums.E.KEY_A: {

### What you expected

I think it should not give any warning? I use a centralised class with most of my Enums, and I get flooded by this when I define dictionaries with these as keys.

### gdstyle version

gdstyle 0.1.7

### Platform

macOS (Apple Silicon)

**Comment by w4nderlust** [comment]
Thanks, this was a genuine bug too. The rule scanned dotted keys token by token, so `Enums.E.KEY_A` became five separate keys and the two dots collided into that bogus `duplicate key '.'`. It now treats the whole key expression as one unit, so enum keys from another class are fine, and real duplicates (including a repeated `Enums.E.KEY_A`) are still flagged.

Fixed in #11, with a follow-up in #12 that also catches duplicates inside nested dictionaries. Both will be in the next release. Thanks for the report.
